openassessment/assessment/test/test_staff.py

```diff
@@ -0,0 +1,331 @@
+from openassessment.assessment.api.peer import create_assessment as peer_assess
+from openassessment.assessment.models import Assessment, PeerWorkflow
+from openassessment.assessment.errors import StaffAssessmentRequestError, StaffAssessmentInternalError
+from openassessment.workflow import api as workflow_api
+from submissions import api as sub_api
+
+@ddt
+class TestStaffAssessment(CacheResetTest):
+    """
+    Tests for staff assessments made as overrides, when none is required to exist.
+    """
+
+    STEP_REQUIREMENTS = {}
+    STEP_REQUIREMENTS_WITH_STAFF = {'staff': {'required': True}}
+
+    # This is due to ddt not playing nicely with list comprehensions
+    ASSESSMENT_SCORES_DDT = [key for key in OPTIONS_SELECTED_DICT]
+
+    @staticmethod
+    @override_settings(ORA2_AI_ALGORITHMS=AI_ALGORITHMS)
+    def _ai_assess(sub):
+        # Note that CLASSIFIER_SCORE_OVERRIDES matches OPTIONS_SELECTED_DICT['most'] scores
+        train_classifiers(RUBRIC, AIGradingTest.CLASSIFIER_SCORE_OVERRIDES)
+        ai_api.on_init(sub, rubric=RUBRIC, algorithm_id=ALGORITHM_ID)
+        return ai_api.get_latest_assessment(sub)
+
+    @staticmethod
+    def _peer_assess(sub, scorer_id, scores):
+        bob_sub, bob = TestStaffAssessment._create_student_and_submission("Bob", "Bob's answer", override_steps=['peer'])
+        peer_api.get_submission_to_assess(bob_sub["uuid"], 1)
+        return peer_assess(bob_sub["uuid"], bob["student_id"], scores, dict(), "", RUBRIC, 1)
+
+    ASSESSMENT_TYPES_DDT = [
+        ('self', lambda sub, scorer_id, scores: self_assess(sub, scorer_id, scores, dict(), "", RUBRIC)),
+        ('peer', lambda sub, scorer_id, scores: TestStaffAssessment._peer_assess(sub, scorer_id, scores)),
+        ('staff', lambda sub, scorer_id, scores: staff_api.create_assessment(sub, scorer_id, scores, dict(), "", RUBRIC)),
+        ('ai', lambda sub, scorer_id, scores: TestStaffAssessment._ai_assess(sub))
+    ]
+
+    @data(*ASSESSMENT_SCORES_DDT)
+    def test_create_assessment_not_required(self, key):
+        """
+        Simple test to ensure staff assessments are scored properly, for all values of OPTIONS_SELECTED_DICT,
+        when staff scores are not required.
+        """
+        # Create assessment
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+
+        # Staff assess it
+        assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            OPTIONS_SELECTED_DICT[key]["options"], dict(), "",
+            RUBRIC,
+        )
+
+        # Ensure points are calculated properly
+        self.assertEqual(assessment["points_earned"], OPTIONS_SELECTED_DICT[key]["expected_points"])
+        self.assertEqual(assessment["points_possible"], RUBRIC_POSSIBLE_POINTS)
+
+        #ensure submission is marked as finished
+        self.assertTrue(staff_api.assessment_is_finished(tim_sub["uuid"], self.STEP_REQUIREMENTS))
```

Comment:
There seems to be a bug on my sandbox where recording a staff score isn't enough for the workflow to say that it is finished. It would be good if this actually verified that the workflow thinks that the whole thing is finished, in addition to the Staff API considering itself finished.